core/uri.c

```diff
@@ -121,6 +122,14 @@ lwm2m_uri_t * lwm2m_decode_uri(multi_option_t *uriPath)
         if (uriPath != NULL) goto error;
         return uriP;
     }
+    else if (URI_DELETE_ALL_SEGMENT_LEN == uriPath->len
+     && 0 == strncmp(URI_DELETE_ALL_SEGMENT, uriPath->data, uriPath->len))
+    {
+        uriP->flag |= LWM2M_URI_FLAG_DELETE_ALL;
+        uriPath = uriPath->next;
+        if (uriPath != NULL) goto error;
+        return uriP;
+    }
```

Comment:
I pused the fix in leshan: https://github.com/eclipse/leshan/commit/aa1e4c492519b0ee378044c7a0840e6e4cfca732